PWG/DevNanoAOD/AliAnalysisNanoAODCuts.cxx

```diff
@@ -734,6 +736,44 @@ Bool_t AliAnalysisNanoAODEventCuts::IsSelected(TObject* obj)
   return kTRUE;
 }
 
+AliAnalysisNanoAODMCParticleCuts::AliAnalysisNanoAODMCParticleCuts()
+    : AliAnalysisCuts(),
+      fDoSelectPrimaries(true),
+      fDoSelectCharged(true),
+      fForceKeepParticles(false),
+      fMinPt(0.f),
+      fMaxEta(999.f),
+      fPDGToKeep(),
+      fPDGV0() {
+}
+
+bool AliAnalysisNanoAODMCParticleCuts::IsSelected(TObject* obj) {
+  // Returns true if the MC particle is to be kept!
+  AliAODMCParticle* part = dynamic_cast<AliAODMCParticle*>(obj);
+
+  // if this is activated the specified particles are kept ignoring the cuts
+  if (fForceKeepParticles) {
+    for (auto it : fPDGToKeep) {
+      if (it == TMath::Abs(part->PdgCode())) {
+        return true;
+      }
+    }
+  }
+  if (part->Pt() < fMinPt) {
+    return false;
+  }
+  if (TMath::Abs(part->Eta()) > fMaxEta) {
+    return false;
+  }
+  if (fDoSelectPrimaries && !part->IsPhysicalPrimary()) {
+    return false;
+  }
+  if (fDoSelectCharged && !part->Charge()) {
+    return false;
```

Comment:
Is the ! clean for comparing to -1 ?

Reply:
yes, it just checks == 0 of != 0
so this should be fine